python/examples/augmented_diff.py

```diff
@@ -113,7 +124,11 @@ def set_old_metadata(elem):
         else:
             obj_id = action.element.get('id')
             if not_in_db(action.element):
-                # TODO verify this is correct
+                # Typically occurs when:
+                #  1. TODO: An element is deleted but then restored later,
```

Comment:
IIUC the OSMExpress database correctly, case 1 will require additional work there to keep the previous state around even after it was deleted (visible set to false) and might bump db size. In this case, version info isn't reset, because the osc file contains current version and metadata info for the element.

Case 2 presents the correct behavior (create operation) as implemented because if we create and then modify an element within the diff window, the element is still created relative to the start time of the window.

Reply:
case 1 sounds like a large increase in scope if we're retaining deleted objects with visible=false indefinitely. 
Isn't it also possible for case 2 to happen on an extract where the program that creates the extract considered the element outside of the extent initially, but then a <modify> appears when it determines it's inside? Again, I haven't tested this extensively in the extract case.

Reply:
> Isn't it also possible for case 2 to happen on an extract where the program that creates the extract considered the element outside of the extent initially, but then a appears when it determines it's inside?

Yes, that would be another case -- I can add it to the list here. You'll also see warnings for incomplete elements when generating diffs for regional extracts when the extract doesn't include child elements outside the extract bbox (I'm having trouble recalling the OSM terminology for this).